### PR TITLE
Added basic project and Cobra CLI structure

### PR DESCRIPTION
**Pull request by surajnarwade**

### CODE REVIEW

File: cmd/app.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	// Here you will define your flags and configuration settings.
+
+	// Cobra supports Persistent Flags which will work for this command
+	// and all subcommands, e.g.:
+	// appCmd.PersistentFlags().String("foo", "", "A help for foo")
+
+	// Cobra supports local flags which will only run when this command
+	// is called directly, e.g.:
+	// appCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
```
Comment (containscafeine) [inline cmd/app.go]: remove these comments?

File: cmd/component.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+	// Here you will define your flags and configuration settings.
+
+	// Cobra supports Persistent Flags which will work for this command
+	// and all subcommands, e.g.:
+	// componentCmd.PersistentFlags().String("foo", "", "A help for foo")
+
+	// Cobra supports local flags which will only run when this command
+	// is called directly, e.g.:
+	// componentCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
```
Comment (containscafeine) [inline cmd/component.go]: remove comments

File: cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+func init() {
+	// Here you will define your flags and configuration settings.
+	// Cobra supports persistent flags, which, if defined here,
+	// will be global for your application.
+	// rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.ocdev.yaml)")
+
+	// Cobra also supports local flags, which will only run
+	// when this action is called directly.
+	rootCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
```
Comment (containscafeine) [inline cmd/root.go]: how about removing this?

File: cmd/version.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	// HEAD is default indicating that this was not set during build
+	GITCOMMIT = "HEAD"
+)
+
+// versionCmd represents the version command
+var versionCmd = &cobra.Command{
+	Use:   "version",
+	Short: "Print the version of ocdev",
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println(VERSION + " (" + GITCOMMIT + ")")
```
Comment (kadel) [inline cmd/version.go]: Do we need this? I've always found annoying rather than useful. But maybe it is just me.

**Pull request merged**